mmcv/utils/config.py

```diff
@@ -199,19 +222,24 @@ def _file2dict(filename, use_predefined_variables=True):
             base_var_dict = Config._pre_substitute_base_vars(
                 temp_config_file.name, temp_config_file.name)
 
+            # Handle base files
+            base_cfg_dict = dict()
+            cfg_text_list = list()
+            for f in Config.parse_base_files(temp_config_file.name):
+                _cfg_dict, _cfg_text = Config._file2dict(osp.join(cfg_dir, f))
+                cfg_text_list.append(_cfg_text)
+                duplicate_keys = base_cfg_dict.keys() & _cfg_dict.keys()
+                if len(duplicate_keys) > 0:
+                    raise KeyError('Duplicate key is not allowed among bases. '
+                                   f'Duplicate keys: {duplicate_keys}')
+                base_cfg_dict.update(_cfg_dict)
+
             if filename.endswith('.py'):
-                temp_module_name = osp.splitext(temp_config_name)[0]
-                sys.path.insert(0, temp_config_dir)
-                Config._validate_py_syntax(filename)
-                mod = import_module(temp_module_name)
-                sys.path.pop(0)
-                cfg_dict = {
-                    name: value
-                    for name, value in mod.__dict__.items()
-                    if not name.startswith('__')
-                }
-                # delete imported module
-                del sys.modules[temp_module_name]
+                cfg_dict = {}
+                with open(temp_config_file.name, 'r') as f:
+                    content = f.read()
+                codeobj = compile(content, '', mode='exec')
+                eval(codeobj, base_cfg_dict, cfg_dict)
```

Comment:
This is a clever trick.

One thing to note is that the eval will introduce many things into the global dict. See the following example. Whether side effects will be caused by these extra variables should be double checked.

![image](https://user-images.githubusercontent.com/9464825/150752698-d3ccc3cc-ea4d-4fc6-9939-7848fdae54d0.png)

Reply:
Right, these interval variables will be removed after merged in to the cfg_dict in line 272-276
```python
        cfg_dict = {
            k: v
            for k, v in cfg_dict.items() if not k.startswith('__')
        }
```

Reply:
Note the `abs/all/any/...`